sdk/core/perf-tests/core-rest-pipeline/test/bearerTokenChallengeAuthenticationPolicy/wwwChallenge.spec.ts

```diff
@@ -0,0 +1,200 @@
+//     Do not use this in production, as values might contain the strings we use to split things up.
+function parseCAEChallenge(challenges: string): any[] {
+  return challenges
+    .split("Bearer ")
+    .filter((x) => x)
+    .map((challenge) =>
+      `${challenge.trim()}, `
+        .split('", ')
+        .filter((x) => x)
+        .map((keyValue) => (([key, value]) => ({ [key]: value }))(keyValue.trim().split('="')))
+        .reduce((a, b) => ({ ...a, ...b }), {})
+    );
+}
+
+async function authorizeRequestOnChallenge(
+  options: AuthorizeRequestOnChallengeOptions
+): Promise<boolean> {
+  const { scopes, request, response, getAccessToken } = options;
+
+  const challenge = response?.headers.get("WWW-Authenticate");
+  if (!challenge) {
+    throw new Error("Failed to retrieve challenge from response headers");
+  }
+
+  const challenges: TestChallenge[] = parseCAEChallenge(challenge) || [];
+
+  const parsedChallenge = challenges.find((x) => x.claims);
+  if (!parsedChallenge) {
+    throw new Error("Missing claims");
+  }
+  if (cachedChallenge !== challenge) {
+    cachedChallenge = challenge;
+  }
+
+  const accessToken = await getAccessToken(scopes, {
+    // The architects haven't decided:
+    // claims: uint8ArrayToString(Buffer.from(parsedChallenge.claims, "base64"))
+  });
+
+  if (!accessToken) {
+    return false;
+  }
+
+  request.headers.set("Authorization", `Bearer ${accessToken}`);
+  return true;
+}
+
+class MockRefreshAzureCredential implements TokenCredential {
+  public authCount = 0;
+  public scopesAndClaims: { scope: string | string[]; challengeClaims: string | undefined }[] = [];
+
+  constructor(public getTokenResponse: AccessToken) {}
+
+  public getToken(
+    scope: string | string[],
+    _options: GetTokenOptions
+  ): Promise<AccessToken | null> {
+    this.authCount++;
+    this.scopesAndClaims.push({
+      scope,
+      // Architects haven't decided about the claims property
+      challengeClaims: undefined // options.claims
+    });
+    return Promise.resolve(this.getTokenResponse);
+  }
+}
+
+export class BearerTokenChallengeAuthenticationPolicyTest extends PerfStressTest {
+  options = {};
+
+  constructor() {
+    super();
+  }
+
+  static pipeline?: Pipeline;
+  static testHttpsClient?: HttpClient;
+  static request?: PipelineRequest;
+
+  async globalSetup(): Promise<void> {
+    const scope = "http://localhost/.default";
+    const challengeClaims = JSON.stringify({
+      access_token: { foo: "bar" }
+    });
+
+    const request = createPipelineRequest({ url: "https://example.com" });
+    const responses: PipelineResponse[] = [
+      {
+        headers: createHttpHeaders({
+          "WWW-Authenticate": `Bearer scope="${scope}", claims="${encodeString(challengeClaims)}"`
+        }),
+        request,
+        status: 401
+      },
+      {
+        headers: createHttpHeaders(),
+        request,
+        status: 200
+      }
+    ];
+
+    const expiresOn = Date.now() + 5000;
```

Comment:
Can the 5000 be parametrized? Is there value in doing it?

Reply:
No value in doing it at the moment, I think 🤔